h2o-core/src/main/java/water/Key.java

```diff
@@ -299,7 +299,37 @@ public static String rand() {
   public static <P extends Keyed> Key<P> makeUserHidden(String s) {
     return make(s,DEFAULT_DESIRED_REPLICA_FACTOR,HIDDEN_USER_KEY, false);
   }
-
+  
+  /** Factory making an random Key from two Strings
+   * 
+   * @param prefix - prefix of the new key. Should not be null. If it contains non-word characters, 
+   *               they are converted to underscores.
+   * @param postfix - postfix of the new key. Should not be null. If it contains non-word characters,
+   *                they are converted to underscores.
+   *
+   *  @return Desired Key that looks like prefix_RANDOM_postfix 
+   */
+  public static <P extends Keyed> Key<P> makeRandom(String prefix, String postfix) {
+    assert prefix != null;
+    assert postfix != null;
+    
+    final String s = prefix + "_" + rand()  + (postfix.isEmpty() ? "" : ("_" + postfix));
+    final String withoutWhitechars = s.replaceAll("\\W", "_");
```

Comment:
Great catch !

Reply:
Honestly, it was not mine, I have seen this in other places in our codebase.